go.mod

```diff
@@ -4,51 +4,53 @@ go 1.21
 
 require (
 	github.com/aws/aws-sdk-go v1.49.13
-	github.com/aws/aws-sdk-go-v2 v1.32.4
-	github.com/aws/aws-sdk-go-v2/config v1.28.4
-	github.com/aws/aws-sdk-go-v2/credentials v1.17.45
-	github.com/aws/aws-sdk-go-v2/service/accessanalyzer v1.36.0
-	github.com/aws/aws-sdk-go-v2/service/acm v1.30.5
-	github.com/aws/aws-sdk-go-v2/service/acmpca v1.37.6
-	github.com/aws/aws-sdk-go-v2/service/amp v1.30.2
-	github.com/aws/aws-sdk-go-v2/service/apigateway v1.27.5
-	github.com/aws/aws-sdk-go-v2/service/apigatewayv2 v1.24.5
-	github.com/aws/aws-sdk-go-v2/service/apprunner v1.32.5
-	github.com/aws/aws-sdk-go-v2/service/autoscaling v1.48.0
-	github.com/aws/aws-sdk-go-v2/service/backup v1.39.6
-	github.com/aws/aws-sdk-go-v2/service/cloudtrail v1.45.0
-	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.43.0
-	github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs v1.43.2
-	github.com/aws/aws-sdk-go-v2/service/codedeploy v1.29.5
-	github.com/aws/aws-sdk-go-v2/service/configservice v1.50.5
-	github.com/aws/aws-sdk-go-v2/service/datasync v1.43.3
-	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.37.0
-	github.com/aws/aws-sdk-go-v2/service/ec2 v1.189.0
+	github.com/aws/aws-sdk-go-v2 v1.32.5
+	github.com/aws/aws-sdk-go-v2/config v1.27.24
+	github.com/aws/aws-sdk-go-v2/credentials v1.17.24
+	github.com/aws/aws-sdk-go-v2/service/accessanalyzer v1.34.2
+	github.com/aws/aws-sdk-go-v2/service/acm v1.30.2
+	github.com/aws/aws-sdk-go-v2/service/acmpca v1.37.3
+	github.com/aws/aws-sdk-go-v2/service/amp v1.27.1
+	github.com/aws/aws-sdk-go-v2/service/apigateway v1.27.2
+	github.com/aws/aws-sdk-go-v2/service/apigatewayv2 v1.24.2
+	github.com/aws/aws-sdk-go-v2/service/apprunner v1.32.2
+	github.com/aws/aws-sdk-go-v2/service/autoscaling v1.23.16
+	github.com/aws/aws-sdk-go-v2/service/backup v1.39.3
+	github.com/aws/aws-sdk-go-v2/service/cloudtrail v1.44.2
+	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.42.2
+	github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs v1.42.0
+	github.com/aws/aws-sdk-go-v2/service/codedeploy v1.29.2
+	github.com/aws/aws-sdk-go-v2/service/configservice v1.50.2
+	github.com/aws/aws-sdk-go-v2/service/datasync v1.42.2
+	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.36.2
+	github.com/aws/aws-sdk-go-v2/service/ec2 v1.184.0
 	github.com/aws/aws-sdk-go-v2/service/ecr v1.36.5
 	github.com/aws/aws-sdk-go-v2/service/ecs v1.49.2
 	github.com/aws/aws-sdk-go-v2/service/efs v1.33.5
-	github.com/aws/aws-sdk-go-v2/service/eks v1.52.0
+	github.com/aws/aws-sdk-go-v2/service/eks v1.51.3
```

Comment:
It seems like you have older package versions cached locally. I prefer to bump all AWS SDK v2 packages before creating the pull request and running unit tests.


```sh
$ cat go.mod|grep -v "//" | grep 'aws-sdk-go-v2' | cut -d " " -f1 | sed 's/[[:space:]]//g' | xargs -L1 go get
$ go mod tidy
```